app/src/main/java/org/mozilla/fenix/settings/DefaultBrowserPreference.kt

```diff
@@ -34,4 +37,18 @@ class DefaultBrowserPreference @JvmOverloads constructor(
         val browsers = Browsers.all(context)
         switchView?.isChecked = browsers.isDefaultBrowser
     }
+
+    fun unsetOpenLinksInAPrivateTabIfNecessary() {
+        if (!Browsers.all(this.context).isDefaultBrowser) {
+            /* If the user has unset us as the default browser, unset openLinksInAPrivateTab */
+            this.context.settings().openLinksInAPrivateTab = false
+            context?.components?.analytics?.metrics?.track(
```

Comment:
Not your fault but I think this metrics tracking is wrong: it says we toggled a preference every time we get an intent and we're not the default browser. If you agree, can you file a bug?

Furthermore, the docs say, "A user toggled a boolean preference in settings". However, this is 1) not being toggled by a user and 2) is not toggled in Settings.

We'll keep the functionality the same in this PR so the Fenix team can address it. Please mention we are changing the code but intend to keep the functionality the same so they don't blame us. :P

Reply:
https://github.com/mozilla-mobile/fenix/issues/7384